test/app-tap/datetime.test.lua

```diff
@@ -1462,6 +1477,21 @@ test:test("Matrix of allowed time and interval additions", function(test)
     test:is(tostring(Y1 + T1970), "1971-01-01T00:00:00Z", "value: Y + T")
     test:is(tostring(Y5 + Y1), "+6 years", "Y + Y")
 
+    -- timezone sensitive operations
+    test:is(tostring(T2000 + I1), "2000-01-02T00:00:00 Europe/Moscow",
+        "value: 2000 + I")
+    test:is(tostring(T2000 + i1), "2000-01-02T00:00:00 Europe/Moscow",
+        "value: 2000 + i")
+    test:is(tostring(T2000 + M2), "2000-03-01T00:00:00 Europe/Moscow",
+        "value: 2000 + 2M")
+    test:is(tostring(T2000 + M2 + M2 + M2), "2000-07-01T00:00:00 Europe/Moscow",
```

Comment:
Please, add some tests/comments regarding timezone offset change.

Reply:
Made it a little bit more verbose, with few comments added here and there:
```diff
diff --git a/test/app-tap/datetime.test.lua b/test/app-tap/datetime.test.lua
index 7367e6522..cb06b0a65 100755
--- a/test/app-tap/datetime.test.lua
+++ b/test/app-tap/datetime.test.lua
@@ -1409,7 +1409,7 @@ Matrix of addition operands eligibility and their result type
 | table           |          |          |          |
 ]]
 test:test("Matrix of allowed time and interval additions", function(test)
-    test:plan(73)
+    test:plan(79)
 
     -- check arithmetic with leap dates
     local T1970 = date.new{year = 1970, month = 1, day = 1}
@@ -1477,20 +1477,41 @@ test:test("Matrix of allowed time and interval additions", function(test)
     test:is(tostring(Y1 + T1970), "1971-01-01T00:00:00Z", "value: Y + T")
     test:is(tostring(Y5 + Y1), "+6 years", "Y + Y")
 
-    -- timezone sensitive operations
-    test:is(tostring(T2000 + I1), "2000-01-02T00:00:00 Europe/Moscow",
+    -- Check winter/DST sensitive operations with intervals.
+    -- We use 2000 year here, because then Moscow still were
+    -- switching between winter and summer time.
+    local msk_offset = 180 -- expected winter time offset
+    local msd_offset = 240 -- expected daylight saving time offset
+
+    local res = T2000 + I1
+    test:is(tostring(res), "2000-01-02T00:00:00 Europe/Moscow",
         "value: 2000 + I")
-    test:is(tostring(T2000 + i1), "2000-01-02T00:00:00 Europe/Moscow",
+    test:is(res.tzoffset, msk_offset, "2000-01-02T00:00:00 - winter")
+
+    res = T2000 + i1
+    test:is(tostring(res), "2000-01-02T00:00:00 Europe/Moscow",
         "value: 2000 + i")
-    test:is(tostring(T2000 + M2), "2000-03-01T00:00:00 Europe/Moscow",
+    test:is(res.tzoffset, msk_offset, "2000-01-02T00:00:00 - winter")
+
+    res = T2000 + M2
+    test:is(tostring(res), "2000-03-01T00:00:00 Europe/Moscow",
         "value: 2000 + 2M")
-    test:is(tostring(T2000 + M2 + M2 + M2), "2000-07-01T00:00:00 Europe/Moscow",
+    test:is(res.tzoffset, msk_offset, "2000-03-01T00:00:00 - winter")
+
+    res = T2000 + M2 + M2 + M2
+    test:is(tostring(res), "2000-07-01T00:00:00 Europe/Moscow",
         "value: 2000 + 6M")
-    test:is(tostring(T2000 + m2), "2000-03-01T00:00:00 Europe/Moscow",
+    test:is(res.tzoffset, msd_offset, "2000-07-01T00:00:00 - summer")
+
+    res = T2000 + m2
+    test:is(tostring(res), "2000-03-01T00:00:00 Europe/Moscow",
         "value: 2000 + 2m")
-    test:is(tostring(T2000 + m2 + M2 + m2), "2000-07-01T00:00:00 Europe/Moscow",
-        "value: 2000 + 6m")
+    test:is(res.tzoffset, msk_offset, "2000-03-01T00:00:00 - winter")
 
+    res = T2000 + m2 + M2 + m2
+    test:is(tostring(res), "2000-07-01T00:00:00 Europe/Moscow",
+        "value: 2000 + 6m")
+    test:is(res.tzoffset, msd_offset, "2000-07-01T00:00:00 - summer")
 
     assert_raises_like(test, expected_interval_but,
                        function() return T1970 + 123 end)
```